File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.java
class Solution {
    public int numberOfArrays(int[] differences, int lower, int upper) {
        long sum1 = 0;
        long min1 = 0;
        long max1 = 0;
        for(int i:differences){
            sum1 += i;
            min1 = Math.min(min1,sum1);
            max1 = Math.max(max1,sum1);
        }
        return (int) Math.max(0, (upper - lower) - (max1 - min1) + 1);
    }
}

File: 2249-count-the-hidden-sequences/2249-count-the-hidden-sequences.py
class Solution:
    def numberOfArrays(self, differences: List[int], lower: int, upper: int) -> int:
        sum1 = 0
        min1 = sum1
        max1 = sum1
        for i in differences:
            sum1 += i
            min1 = min(min1,sum1)
            max1 = max(max1,sum1)
        ub = upper - max1
        lb = lower - min1
        if( ub-lb+1>0):
            return ub-lb+1
        return 0